docs/migration_guide.md

```diff
@@ -75,11 +87,27 @@ In Mesa 3.0, `unique_id` for agents is now automatically assigned, simplifying a
 
 
 ### AgentSet and `Model.agents`
-#### AgentSet
-<!-- TODO  -->
+In Mesa 3.0, the Model class internally manages agents using several data structures:
+
+- `self._agents`: A dictionary containing hard references to all agents, indexed by their `unique_id`.
+- `self._agents_by_type`: A dictionary of AgentSets, organizing agents by their type.
+- `self._all_agents`: An AgentSet containing all agents in the model.
+
+These internal structures are used to efficiently manage and access agents. Users should interact with agents through the public `model.agents` property, which returns the `self._all_agents` AgentSet.
```

Comment:
If I want to scrutinize an agent from `model.agents`, could be for data collection, but more often for debugging purpose, how would I do this efficiently? Currently, it is doable with `select(at_most=1)`, but this returns yet another `AgentSet`. In Polars, I can do `df.row(2)`.

Reply:
You can just do list-like `agentset[2]`

Reply:
I didn't see there is `__getitem__` documented in the public API: https://mesa.readthedocs.io/en/latest/apis/agent.html. This feature is documented in the code, but doesn't show up there.

Reply:
What if I want to access by a unique ID instead of the index? (iloc vs loc in pandas)

Reply:
All good questions, but I don’t think they are relevant to this PR.

Maybe open an issue or discussion about this?

Reply:
It's relevant in the sense that the user may have to do such thing in their existing code.

Reply:
I don’t exactly understand what you want added / modified, so could you open an PR to add it?